ros2caret/verb/create_architecture.py

```diff
@@ -66,23 +67,53 @@ def add_arguments(self, parser, cli_name):
             help='allow overwrite of architecture file',
             required=False, default=False
         )
+        parser.add_argument(
+            '-m', '--max_callback_construction_order_on_path_searching',
+            type=int, dest='max_callback_construction_order_on_path_searching',
+            help='callbacks whose construction_order are greater than'
+            ' this value are ignored on path searching.'
+            ' The value must be positive integer or "0". "0" means unlimited.'
+            ' Default: %(default)s',
+            required=False, default=DEFAULT_MAX_CALLBACK_CONSTRUCTION_ORDER_ON_PATH_SEARCHING,
+        )
 
     def main(self, *, args):
-        create_arch = CreateArchitecture(args.trace_dir)
-        create_arch.create(args.output_path, args.force)
+        try:
+            create_arch = CreateArchitecture(
+                args.trace_dir,
+                args.max_callback_construction_order_on_path_searching
+            )
+            create_arch.create(args.output_path, args.force)
+        except Exception as e:
+            logger.warning(e)
+            return 1
+        return 0
 
 
 class CreateArchitecture:
 
     def __init__(
         self,
         trace_dir: str,
+        max_callback_construction_order_on_path_searching: int,
         architecture: Optional[Architecture] = None
     ) -> None:
         if architecture:
             self._arch = architecture
         else:
-            self._arch = Architecture('lttng', trace_dir)
+            if max_callback_construction_order_on_path_searching >= 0:
+                tmp = max_callback_construction_order_on_path_searching
+                self._arch = Architecture(
+                    'lttng',
+                    trace_dir,
+                    max_callback_construction_order_on_path_searching=tmp
+                )
```

Comment:
I suggest removing `tmp`.

```suggestion
                self._arch = Architecture(
                    'lttng',
                    trace_dir,
                    max_callback_construction_order_on_path_searching
                )
```

Reply:
consent.